webapp/src/components/common/dialogs/DataViewerDialog/index.tsx

```diff
@@ -23,7 +26,9 @@ interface Props {
 function DataViewerDialog(props: PropsWithChildren<Props>) {
   const [t] = useTranslation();
   const { enqueueSnackbar } = useSnackbar();
-  const { data, onClose, isMatrix } = props;
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+  const { studyId, data, onClose, isMatrix } = props;
+  const [matrixIndex, setMatrixIndex] = useState<MatrixIndex>();
 
   const copyId = (matrixId: string): void => {
```

Comment:
Use async/await for `navigator.clipboard.writeText(matrixId);` that returns a promise.